specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/assessments.json

```diff
@@ -0,0 +1,246 @@
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SecurityAssessmentProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "SecurityAssessmentProperties": {
+      "type": "object",
+      "description": "Describes properties of an assessment.",
+      "properties": {
+        "resourceDetails": {
+          "$ref": "#/definitions/ResourceDetails"
+        },
+        "displayName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "User friendly display name of the assessment"
+        },
+        "status": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Status of the ",
+          "enum": ["Passed", "Failed", "NotApplicable"],
+          "x-ms-enum": {
+            "name": "Severity",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Passed",
+                "description": "The resource is healthy"
+              },
+              {
+                "value": "Failed",
+                "description": "The resource has a security issue that needs to be addressed"
+              },
+              {
+                "value": "NotApplicable",
+                "description": "Assessment for this resource did not happen"
+              }
+            ]
+          }
+        },
+        "notApplicableReasonCode": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Programmatic code for the reason the assessment result is NotApplicable"
+        },
+        "notApplicableReasonDescription": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Human readable description for the reason the assessment result is NotApplicable"
+        },
+        "assessedResourceSource": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The platform where the assessed resource resides"
+        },
+        "additionalData": {
+          "readOnly": true,
+          "type": "array",
+          "description": "Additional data regarding the assessment",
+          "items":{
+            "$ref": "#/definitions/AssessmentAdditionalData"
+          }
+        }
+      }
+    },
+    "AssessmentAdditionalData": {
+      "type": "object",
+      "description": "Additional data regarding the assessment",
+      "properties": {
+        "key": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Name of the field of the additional data"
+        },
+        "value": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Value of the field of the additional data"
+        }
+      }
+    },
+    "ResourceDetails": {
+      "type": "object",
+      "description": "Details of the resource that was assessed"
+    },
+    "AzureResourceDetails": {

```

Comment:
I don't even see this type referenced, is it required?  If so can you please provide an example of the JSON that's sent over the wire?

Reply:
it is an inheritance of the "ResourceDetails" object (which is like an abstract class)
I added the discriminator property, hope that now the documentation is better